### PR TITLE
#2567 - update donor summary sorting

### PR DESCRIPTION
**Pull request by samrichca**
# Description of changes

- small refactor of donor summary table sorting to add types and reduce duplication
- update comments

## Type of Change

- [ ] Bug
- [ ] Dependency updates
- [ ] Feature
- [x] Refactoring
- [ ] Release candidate
- [ ] Styling
- [ ] Testing
- [ ] Other (please describe)

## Checklist before requesting review

- Design (select one):
  - [ ] Matches design:
    - component sizes, spacing, and styles
    - font size, weight, colour
    - spelling has been double checked
  - [ ] No design provided, screenshot of changes included in PR
  - [x] No changes to UI
- Back-end (select one):
  - [ ] Back-end PRs merged, tested on develop, and linked in this PR
  - [x] No back-end changes
- [x] Manually tested changes
- [x] Added copyrights to new files
- [x] Connected ticket to PR

### CODE REVIEW

File: components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx [hunk 1]
```diff
@@ -83,22 +88,23 @@ const PIPELINE_COLORS: { [k: string]: keyof ThemeColorNames } = {
...
+const MULTI_SORT_COLUMN_IDS: Partial<DonorSummaryMultiSortColumnIds> = {
+  alignmentsCompleted: ['alignmentsRunning', 'alignmentsFailed'],
+  mutectCompleted: ['mutectRunning', 'mutectFailed'],
+  openAccessCompleted: ['openAccessRunning', 'openAccessFailed'],
+  publishedNormalAnalysis: ['publishedTumourAnalysis'],
+  registeredNormalSamples: ['registeredTumourSamples'],
+  rnaAlignmentsCompleted: ['rnaAlignmentsRunning', 'rnaAlignmentFailed'],
+  rnaPublishedNormalAnalysis: ['rnaPublishedTumourAnalysis'],
+  rnaRegisteredNormalSamples: ['rnaRegisteredTumourSamples'],
+  sangerVcsCompleted: ['sangerVcsRunning', 'sangerVcsFailed'],
```
Comment (ciaranschutte) [inline components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx:100]: with some vars using UPPERCASE_SNAKE_CASE are they meant to be frozen objects or typescript constants?

**Review by ciaranschutte — APPROVED** [review]
suggestions not request for changes

File: components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx [hunk 2]
```diff
@@ -135,7 +141,7 @@ const DonorSummaryTable = ({
     const newSort = getSorts();
     const sorts = newSort.reduce(
       (accSorts: Array<DonorSummarySortingState>, sortRule: DonorSummarySortingState) => {
-        const fields = MULTI_SORT_COLUMN_IDS[sortRule.id] || [sortRule.id];
+        const fields = [sortRule.id, ...(MULTI_SORT_COLUMN_IDS[sortRule.id] || [])];
```
[the remark this answers is not in the document]
Comment (samrichca) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx:144]: like `[sortRule.id].concat(MULTI_SORT_COLUMN_IDS[sortRule.id] || [])`?

[the remark this answers is not in the document]
Comment (ciaranschutte) [reply components/pages/submission-system/program-dashboard/DonorDataSummary/DonorSummaryTable.tsx:144]: like:`[sortRule.id].concat(MULTI_SORT_COLUMN_IDS[sortRule.id])` concat(undefined) won't added anything

**Pull request merged**